From ddaa8489845083d6813f108e93c94fd1c4cb94cf Mon Sep 17 00:00:00 2001
From: Greg Scaffidi <sgscaffidi3@gmail.com>
Date: Fri, 4 Apr 2025 20:37:44 -0500
Subject: [PATCH] fix linking

---
 Makefile.am  | 1 +
 configure.ac | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 261a58d..13bd2d7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -57,6 +57,7 @@ include_HEADERS = $(top_srcdir)/include/ffms.h $(top_srcdir)/include/ffmscompat.
 bin_PROGRAMS = src/index/ffmsindex
 src_index_ffmsindex_SOURCES = src/index/ffmsindex.cpp
 src_index_ffmsindex_LDADD = -lavutil src/core/libffms2.la
+src_index_ffmsindex_LDFLAGS = @src_index_ffmsindex_LDFLAGS@
 
 .PHONY: test test-build test-clean test-sync test-run
 clean-local: test-clean
diff --git a/configure.ac b/configure.ac
index b875b30..a4c6e49 100644
--- a/configure.ac
+++ b/configure.ac
@@ -150,8 +150,10 @@ AC_MSG_RESULT($bsymbolic)
 
 if echo "$host" | $GREP "mingw" >/dev/null 2>&1; then
     LTUNDEF="-no-undefined"
+    src_index_ffmsindex_LDFLAGS="$src_index_ffmsindex_LDFLAGS -municode"
 fi
 AC_SUBST([LTUNDEF])
+AC_SUBST([src_index_ffmsindex_LDFLAGS])
 
 AC_CONFIG_FILES([
 Makefile
-- 
2.45.2.windows.1

