From cf824464cce21c5dce63ea5a08d66f9cf54af179 Mon Sep 17 00:00:00 2001
From: Greg Scaffidi <sgscaffidi3@gmail.com>
Date: Fri, 4 Apr 2025 19:07:44 -0500
Subject: [PATCH] fix linking

---
 Makefile.am  | 1 +
 configure.ac | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 261a58d..13bd2d7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -57,6 +57,7 @@ include_HEADERS = $(top_srcdir)/include/ffms.h $(top_srcdir)/include/ffmscompat.
 bin_PROGRAMS = src/index/ffmsindex
 src_index_ffmsindex_SOURCES = src/index/ffmsindex.cpp
 src_index_ffmsindex_LDADD = -lavutil src/core/libffms2.la
+src_index_ffmsindex_LDFLAGS = @src_index_ffmsindex_LDFLAGS@
 
 .PHONY: test test-build test-clean test-sync test-run
 clean-local: test-clean
diff --git a/configure.ac b/configure.ac
index b875b30..5607f8f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -137,7 +137,7 @@ if test "$enable_shared" = yes; then
         bsymbolic=no
     fi
     LDFLAGS="$_LDFLAGS"
-    src_core_libffms2_la_LDFLAGS="$src_core_libffms2_la_LDFLAGS -version-info $VERSION_INFO"
+    src_core_libffms2_la_LDFLAGS="$src_core_libffms2_la_LDFLAGS -version-info $VERSION_INFO -municode"
 else
     bsymbolic=no
 fi
@@ -152,6 +152,7 @@ if echo "$host" | $GREP "mingw" >/dev/null 2>&1; then
     LTUNDEF="-no-undefined"
 fi
 AC_SUBST([LTUNDEF])
+AC_SUBST([src_index_ffmsindex_LDFLAGS])
 
 AC_CONFIG_FILES([
 Makefile
-- 
2.45.2.windows.1

